### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by tkreuzer**
This adds a CODEOWNERS file.
The difference to the MAINTAINERS file is, that GitHub supports it to add reviewers and optionally require signoffs. See: https://help.github.com/en/articles/about-code-owners

This is a first draft, where I added some pieces from the MAINTAINERS file, trying to keep the same format.

I also added an entry using a GitHub team (lang-german for german resource files)
See https://help.github.com/en/articles/about-teams

Example: https://github.com/python/cpython/blob/master/.github/CODEOWNERS
Git extension: https://github.com/softprops/git-codeowners

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+#	S: Status, one of the following:
+#	   Maintained:  Someone is handling that area
+#	   Upstream: This is 3rd party code, synced in our tree
+#	   Abandoned:  No one is handling that code anymore
+#	F: Files. Directories, files (wildcards allowed) covered in
+#	   this area
+#	C: Comments
+
+
+# 3rd Party File Format Libraries
```
Comment (learn-more) [inline CODEOWNERS]: I would suggest to make a distinction between these functional comments, and the block above explaining stuff.
For example, use double `##` comment markers for the comment block.
This greatly simplifies parsing this file to create a script like `get_maintainer.py`

Comment (learn-more) [reply CODEOWNERS]: Or prefix the lines that should be read by the script with `#>` or something like that.

Comment (tkreuzer) [reply CODEOWNERS]: We can also install the git extension, I linked in the description. I'm not too familiar with how that works and whether we can do that once for everyone, though.

**Comment by HBelusca** [comment]
I don't know, whether it could be possible to have a script that would generate the CODEOWNERS file from the (more complete?) MAINTAINERS file @HeisSpiter added already.

**Comment by ColinFinck** [comment]
The pedantic in me would replace the tabs between path name and maintainer by a lot of spaces to align properly, or alternatively by 2-4 spaces and not even try to align all maintainers, because the length of a path name is hardly predictable.
Otherwise, looks good to me!

If CODEOWNERS is supported by GitHub while MAINTAINERS is not, I think this file should replace MAINTAINERS as soon as possible. There is no point in maintaining two of them.

**Comment by HeisSpiter** [comment]
Totally agreeing with Colin's message.
I think this PR should contain the full CODEOWNERS files and MAINTAINERS file deletion.

**Pull request merged**